server/src/resources/js/checkout.ts

```diff
@@ -0,0 +1,233 @@
+
+                            setTimeout(() => {
+                                console.log('redirect');
+                                try {
+                                    let eventData: [{ data: { redirect_url: string } }] = JSON.parse(
+                                        re.data.checkout.event_data,
+                                    );
+                                    let redirectUrl = eventData[0].data.redirect_url;
+
+                                    localStorage.setItem('event_data', JSON.stringify(eventData));
+                                    window.location.replace(redirectUrl);
+                                } catch (error) {
+                                    console.log(error);
+                                }
+                            }, 4000);
+                        }
+                    });
+                },
+            };
+        };
+    }
+
+    public sendOfCheckoutInfo(values: Gernzy.CheckoutInfo) {
+        var userToken = localStorage.getItem('userToken') || '';
+        let query = ` mutation {
+            checkout(input: {
+                name: "${values['name']}",
+                email: "${values['email']}",
+                telephone: "${values['telephone']}",
+                mobile: "${values['mobile']}",
+                billing_address: {
+                    line_1: "${values['billing_address_line_1']}",
+                    line_2: "${values['billing_address_line_2']}",
+                    state: "${values['billing_address_state']}",
+                    postcode: "${values['billing_address_postcode']}",
+                    country: "${values['billing_address_country']}"
+                },
+                shipping_address: {
+                    line_1: "${values['shipping_address_line_1']}",
+                    line_2: "${values['shipping_address_line_2']}",
+                    state: "${values['shipping_address_state']}",
+                    postcode: "${values['shipping_address_postcode']}",
+                    country: "${values['shipping_address_country']}"
+                },
+                use_shipping_for_billing: ${values['use_shipping_for_billing']},
+                payment_method: "${values['payment_method']}",
+                agree_to_terms: ${values['agree_to_terms']},
+                notes: "${values['notes']}"
+            }){
+                event_data
+                order {
+                    id
+                }
+            }
+        }`;
+
+        return this.graphqlService.sendQuery(query, userToken, this.url);
+    }
+
+    public getBasketTotal() {
+        var userToken = localStorage.getItem('userToken') || '';
+        var self = this;
+
+        let query = `{
+            me {
+                cart {
+                    cart_total
+                }
+            }
+        }`;
+
+        window.checkoutCartTotal = () => {
```

Comment:
We can apply the public interface here as well, e.g. `window.checkoutCartTotal = this.makePublicInterface()`

Reply:
completed